velox/exec/benchmarks/PrefixSortBenchmark.cpp

```diff
@@ -0,0 +1,236 @@
+    if (rowType->children().size() == 1) {
+      return fuzzer.fuzzRow(rowType);
+    }
+    std::vector<VectorPtr> children;
+    for (int column = 0; column < rowType->children().size() - 1; ++column) {
+      children.push_back(fuzzerWithNulls.fuzz(rowType->childAt(column)));
+    }
+    children.push_back(
+        fuzzer.fuzz(rowType->childAt(rowType->children().size() - 1)));
+    return std::make_shared<RowVector>(
+        pool, rowType, nullptr, numRows, std::move(children));
+  }
+
+  const std::string testName_;
+  const size_t numRows_;
+  // Rows address stored in RowContainer
+  std::vector<char*> rows_;
+  std::unique_ptr<RowContainer> data_;
+};
+
+class PrefixSortBenchmark {
+ public:
+  PrefixSortBenchmark() : pool_(rootPool()->addLeafChild("PrefixSortTest")) {}
+
+  void runPrefixSort(
+      const std::vector<char*>& rows,
+      RowContainer* rowContainer,
+      const std::vector<CompareFlags>& compareFlags) {
+    PrefixSortConfig prefixSortConfig(1024);
+    PrefixSort prefixSort(
+        pool_.get(), rowContainer, compareFlags, prefixSortConfig);
+    // Copy rows to sortedRows to avoid sort rows already sorted.
+    std::vector<char*> sortedRows = rows;
+    prefixSort.sort(sortedRows);
+  }
+
+  void runStdSort(
+      const std::vector<char*>& rows,
+      RowContainer* rowContainer,
+      const std::vector<CompareFlags>& compareFlags) {
+    std::vector<char*> sortedRows = rows;
+    std::sort(
+        sortedRows.begin(),
+        sortedRows.end(),
+        [&](const char* leftRow, const char* rightRow) {
+          for (vector_size_t index = 0; index < compareFlags.size(); ++index) {
+            if (auto result = rowContainer->compare(
+                    leftRow, rightRow, index, compareFlags[index])) {
+              return result < 0;
+            }
+          }
+          return false;
+        });
+  }
+
+  // Add benchmark manually to avoid writing a lot of BENCHMARK.
+  void benchMark(
+      const std::string& testName,
+      size_t numRows,
+      const RowTypePtr& rowType,
+      int iterateTimes = 1) {
+    // CompareFlags could be same in benchmark.
+    std::vector<CompareFlags> compareFlags;
+    for (int i = 0; i < rowType->children().size(); ++i) {
+      compareFlags.push_back(
+          {true, true, false, CompareFlags::NullHandlingMode::kNullAsValue});
+    }
+    auto testCase =
+        std::make_unique<TestCase>(pool_.get(), testName, numRows, rowType);
+    // Add benchmarks for std-sort and prefix-sort.
+    {
+      folly::addBenchmark(
+          __FILE__,
+          "StdSort_" + testCase->testName(),
+          [rows = testCase->rows(),
+           container = testCase->rowContainer(),
+           sortFlags = compareFlags,
+           iterateTimes = iterateTimes,
+           this]() {
+            for (auto i = 0; i < iterateTimes; ++i) {
+              runStdSort(rows, container, sortFlags);
+            }
+            return 1;
```

Comment:
I'm thinking that it is interesting to compare performance per row. To achieve that we may need to return rows.size() * iterateTimes here instead of 1.

Here is what this would look like in the results.

```
============================================================================
[...]ec/benchmarks/PrefixSortBenchmark.cpp     relative  time/iter   iters/s
============================================================================
StdSort_1_Bigint_1k                                       155.86ns     6.42M
PrefixSort_1_Bigint_1k                          88.714%   175.69ns     5.69M
StdSort_2_Bigint_1k                                       227.63ns     4.39M
PrefixSort_2_Bigint_1k                          120.95%   188.20ns     5.31M
StdSort_3_Bigint_1k                                       292.17ns     3.42M
PrefixSort_3_Bigint_1k                          160.41%   182.14ns     5.49M
StdSort_4_Bigint_1k                                       326.12ns     3.07M
PrefixSort_4_Bigint_1k                          189.11%   172.45ns     5.80M
StdSort_1_Bigint_10k                                      226.20ns     4.42M
PrefixSort_1_Bigint_10k                         105.70%   214.00ns     4.67M
StdSort_2_Bigint_10k                                      372.98ns     2.68M
PrefixSort_2_Bigint_10k                         160.16%   232.88ns     4.29M
StdSort_3_Bigint_10k                                      495.61ns     2.02M
PrefixSort_3_Bigint_10k                         218.71%   226.60ns     4.41M
StdSort_4_Bigint_10k                                      637.86ns     1.57M
PrefixSort_4_Bigint_10k                         306.07%   208.40ns     4.80M
StdSort_1_Bigint_100k                                     317.01ns     3.15M
PrefixSort_1_Bigint_100k                        130.84%   242.29ns     4.13M
StdSort_2_Bigint_100k                                     488.82ns     2.05M
PrefixSort_2_Bigint_100k                        189.78%   257.57ns     3.88M
StdSort_3_Bigint_100k                                     616.25ns     1.62M
PrefixSort_3_Bigint_100k                        219.56%   280.67ns     3.56M
StdSort_4_Bigint_100k                                     734.61ns     1.36M
PrefixSort_4_Bigint_100k                        304.56%   241.21ns     4.15M
```